pkg/plugin/state.go

```diff
@@ -686,20 +686,30 @@ func (e *AutoscaleEnforcer) reserveResources(
 			MqIndex:        -1,
 			MigrationState: nil,
 		}
+		// initially build the resource states assuming that we're including buffer, and then update
+		// later to remove it if that turns out not to be right.
 		cpuState = podResourceState[vmapi.MilliCPU]{
-			Reserved:         vmInfo.Using().VCPU,
-			Buffer:           0,
+			Reserved:         vmInfo.Max().VCPU,
+			Buffer:           util.SaturatingSub(vmInfo.Max().VCPU, vmInfo.Using().VCPU),
 			CapacityPressure: 0,
 			Min:              vmInfo.Min().VCPU,
 			Max:              vmInfo.Max().VCPU,
 		}
 		memState = podResourceState[api.Bytes]{
-			Reserved:         vmInfo.Using().Mem,
-			Buffer:           0,
+			Reserved:         vmInfo.Max().Mem,
+			Buffer:           util.SaturatingSub(vmInfo.Max().Mem, vmInfo.Using().Mem),
 			CapacityPressure: 0,
 			Min:              vmInfo.Min().Mem,
 			Max:              vmInfo.Max().Mem,
 		}
+
+		migrating := util.TryPodOwnerVirtualMachineMigration(pod) != nil
+		if !vmInfo.Config.ScalingEnabled || migrating {
```

Comment:
When Buffer values are zeroed if ScalingEnabled=true? Also, these two lines had a comment before, maybe bring it along?

Reply:
> When Buffer values are zeroed if ScalingEnabled=true?

Not sure what you're asking, could you rephrase?

> Also, these two lines had a comment before, maybe bring it along?

Ah good catch, thanks! Fixed in b3f2d70a764418b44ce502b81e496cf1b5084322

Reply:
> > When Buffer values are zeroed if ScalingEnabled=true?
> 
> Not sure what you're asking, could you rephrase?

If ScalingEnabled=true, then buffer values are non-zero. How do they become zero? When we hear the report from the autoscaler-agent?

Reply:
Correct, yeah - when handling the first autoscaler-agent request.